src/messaging/messages/response-message.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * @module ResponseMessage
+ */
+import Message from './message.js'
+
+/** A response message that is sent as an answer to the request message. */
+export default class ResponseMessage extends Message {
+  /**
+   * @param {RequestMessage} request - A request that initiated this response. Used to copy routing information mostly.
+   * @param {object} body - A body of the response, a plain JS object with no methods.
+   * @param {string} responseCode - A code to indicate results of the request handling: Success, Failure, etc.
+   */
+  constructor (request, body = {}, responseCode = ResponseMessage.responseCodes.UNDEFINED) {
+    super(body)
+    this.role = Message.roles.RESPONSE
+    this.requestHeader = request.header || {}
```

Comment:
A header of a request that initiated the response is copied into the response message object. It can be useful for tracking from where and in what form did the request come from.